Fluidsynth doesn't compile with `-Dosal=cpp11 -Denable-libinstpatch=0` when targetting for macos < 10.15

### FluidSynth version

```sh
2.5.0
```

### Describe the bug

While I see that there is detection code at https://github.com/FluidSynth/fluidsynth/blob/f0c69790df4598fbb80cb82192da1fd97ddbc879/CMakeLists.txt#L566-L574

It doesn't work on macos as intended because cmake finds the `filesystem` module (setting `HAVE_CXX_FILESYSTEM=1`), but later during the compilation there's an error.

### Expected behavior

Internally `HAVE_CXX_FILESYSTEM` is set to `0` and compilation succeeds with the stub functions. If this is not possible, export some public cmake option to override detection and allow users to set `HAVE_CXX_FILESYSTEM`.

### Steps to reproduce

compile fluidsynth with `MACOSX_DEPLOYMENT_TARGET` env set to some older version (while also targetting for intel macs)

### Additional context

Full compilation error log

```
+ cmake .. -DCMAKE_INSTALL_PREFIX=[...] -DCMAKE_INSTALL_LIBDIR:PATH=lib -DCMAKE_BUILD_TYPE=Release -DBUILD_SHARED_LIBS=true -DCMAKE_INSTALL_NAME_DIR=[...] -DCMAKE_PREFIX_PATH=[...] -DCMAKE_OSX_ARCHITECTURES=x86_64 -Denable-readline=OFF -Denable-framework=NO -Denable-openmp=OFF -Dosal=cpp11 -Denable-libinstpatch=0 -Denable-pulseaudio=NO -Denable-pipewire=NO
-- The C compiler identification is AppleClang 17.0.0.17000013
-- The CXX compiler identification is AppleClang 17.0.0.17000013
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/clang - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/clang++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Checking whether system has ANSI C header files
-- Looking for 8 include files dlfcn.h, ..., float.h
-- Looking for 8 include files dlfcn.h, ..., float.h - found
-- Performing Test memchrExists
-- Performing Test memchrExists - Success
-- Performing Test freeExists
-- Performing Test freeExists - Success
-- ANSI C header files - found
-- Looking for include file unistd.h
-- Looking for include file unistd.h - found
-- Looking for DIR in sys/stat.h;sys/types.h;dirent.h
-- Looking for DIR in sys/stat.h;sys/types.h;dirent.h - found
-- Looking for stdio.h
-- Looking for stdio.h - found
-- Looking for math.h
-- Looking for math.h - found
-- Looking for errno.h
-- Looking for errno.h - found
-- Looking for stdarg.h
-- Looking for stdarg.h - found
-- Looking for sys/mman.h
-- Looking for sys/mman.h - found
-- Looking for sys/time.h
-- Looking for sys/time.h - found
-- Looking for fcntl.h
-- Looking for fcntl.h - found
-- Looking for sys/socket.h
-- Looking for sys/socket.h - found
-- Looking for netinet/in.h
-- Looking for netinet/in.h - found
-- Looking for netinet/tcp.h
-- Looking for netinet/tcp.h - found
-- Looking for arpa/inet.h
-- Looking for arpa/inet.h - found
-- Looking for limits.h
-- Looking for limits.h - found
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for signal.h
-- Looking for signal.h - found
-- Looking for getopt.h
-- Looking for getopt.h - found
-- Check size of long long
-- Check size of long long - done
-- Performing Test _have_inline
-- Performing Test _have_inline - Success
-- Performing Test _have_vla
-- Performing Test _have_vla - Success
-- Performing Test HAVE_INCOMPATIBLE_POINTER_TYPES
-- Performing Test HAVE_INCOMPATIBLE_POINTER_TYPES - Success
The 'gcem' submodule directory seems to be empty or incomplete. Attempting to download gcem from Github.
-- [download 8% complete]
-- [download 17% complete]
-- [download 24% complete]
-- [download 33% complete]
-- [download 42% complete]
-- [download 51% complete]
-- [download 59% complete]
-- [download 68% complete]
-- [download 77% complete]
-- [download 86% complete]
-- [download 95% complete]
-- [download 100% complete]
-- gcem source downloaded and extracted to [...]/fluidsynth-2.5.0/build/gcem-extracted
-- Found GCEM: [...]/fluidsynth-2.5.0/build/gcem-extracted/gcem-012ae73c6d0a2cb09ffe86475f5c6fba3926e200/include
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for CoreAudio/AudioHardware.h
-- Looking for CoreAudio/AudioHardware.h - found
-- Looking for CoreMIDI/MIDIServices.h
-- Looking for CoreMIDI/MIDIServices.h - found
-- Looking for C++ include filesystem
-- Looking for C++ include filesystem - found
-- Found libSndFile: 1.2.2
-- SndFile_WITH_EXTERNAL_LIBS="1" ; SndFile_WITH_MPEG="0"
-- Could NOT find ALSA (missing: ALSA_LIBRARY ALSA_INCLUDE_DIR) (Required is at least version "0.9.1")
-- Could NOT find Jack (missing: Jack_LIBRARY Jack_INCLUDE_DIR) 
-- Could NOT find DBus1 (Set DBus1_DIR to the directory containing its CMake config)
-- Looking for ladspa.h
-- Looking for ladspa.h - not found
-- Looking for sys/soundcard.h
-- Looking for sys/soundcard.h - not found
-- Looking for linux/soundcard.h
-- Looking for linux/soundcard.h - not found
-- Looking for machine/soundcard.h
-- Looking for machine/soundcard.h - not found
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES) 
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND C) 
-- Looking for sinf
-- Looking for sinf - found
-- Looking for cosf
-- Looking for cosf - found
-- Looking for fabsf
-- Looking for fabsf - found
-- Looking for powf
-- Looking for powf - found
-- Looking for sqrtf
-- Looking for sqrtf - found
-- Looking for logf
-- Looking for logf - found
-- Looking for inet_ntop
-- Looking for inet_ntop - found
-- Check size of socklen_t
-- Check size of socklen_t - done
-- Could NOT find Doxygen (missing: DOXYGEN_EXECUTABLE) 
-- Current revision (Git) is f44ef3e
**************************************************************
-- 

**************************************************************

Build Summary:
FluidSynth Version:    2.5.0
Library version:       3.4.0
Git revision:          f44ef3e
Build type:            Release
Install Prefix:        [...]

Audio / MIDI driver support:
  ALSA:                  no
  CoreAudio:             yes
  CoreMIDI:              yes
  DSound:                no
  JACK:                  no
  MidiShare:             no
  Oboe:                  no
  OpenSLES:              no
  OS/2 DART:             no
  OS/2 KAI:              no
  OSS:                   no
  PipeWire:              no
  PortAudio:             no
  PulseAudio:            no
  SDL3:                  no
  WASAPI:                no
  WaveOut:               no
  WinMidi:               no

Support for SF3 files:   yes
Support for DLS files:   yes
  native:                yes
  libinstpatch:          no

Audio to file rendering: yes
  libsndfile:            yes

Miscellaneous support:
  D-Bus:                 no
  LADSPA support:        no
  NETWORK Support:       yes
    IPV6 Support:        yes
  Readline:              no
  systemd:               no
  getopt:                yes

Developer nerds info:
  OS abstraction:        cpp11
  Samples type:          double
  Multithread rendering: yes
  OpenMP 4.0:            no
  Profiling:             no
  Debug Build:           no
  Trap on FPE (debug):   no
  Check FPE (debug):     no
  UBSan (debug):         no
  Coverage:              no

-- Configuring done (9.0s)
-- Generating done (0.2s)
+ make
-- Build files have been written to: [...]/fluidsynth-2.5.0/build
[  2%] Building C object src/CMakeFiles/libfluidsynth-OBJ.dir/drivers/fluid_aufile.c.o
[  5%] Building C object src/CMakeFiles/libfluidsynth-OBJ.dir/drivers/fluid_coremidi.c.o
[  5%] Building C object src/CMakeFiles/libfluidsynth-OBJ.dir/drivers/fluid_coreaudio.c.o
[  8%] Building CXX object src/CMakeFiles/libfluidsynth-OBJ.dir/sfloader/fluid_dls.cpp.o
[ 10%] Building CXX object src/CMakeFiles/libfluidsynth-OBJ.dir/utils/fluid_sys_cpp11.cpp.o
[ 10%] Building CXX object src/CMakeFiles/libfluidsynth-OBJ.dir/utils/fluid_file.cpp.o
[ 13%] Building CXX object src/CMakeFiles/libfluidsynth-OBJ.dir/gentables/fluid_ct2hz.cpp.o
[...]/fluidsynth-2.5.0/src/utils/fluid_sys_cpp11.cpp:274:26: error: 'path' is unavailable: introduced in macOS 10.15 unknown
  274 |         std::filesystem::path path = std::filesystem::u8path(_path);
      |                          ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:382:33: note: 'path' has been explicitly marked unavailable here
  382 | class _LIBCPP_EXPORTED_FROM_ABI path {
      |                                 ^
[...]/fluidsynth-2.5.0/src/utils/fluid_sys_cpp11.cpp:274:55: error: 'u8path<const char *, 0>' is unavailable: introduced in macOS 10.15 unknown
[ 16%] Building CXX object src/CMakeFiles/libfluidsynth-OBJ.dir/gentables/fluid_cb2amp.cpp.o
  274 |         std::filesystem::path path = std::filesystem::u8path(_path);
      |                                                       ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/u8path.h:78:60: note: 'u8path<const char *, 0>' has been explicitly marked unavailable here
   78 | _LIBCPP_HIDE_FROM_ABI _LIBCPP_DEPRECATED_WITH_CHAR8_T path u8path(const _Source& __s) {
      |                                                            ^
[...]/fluidsynth-2.5.0/src/utils/fluid_sys_cpp11.cpp:274:38: error: '~path' is unavailable: introduced in macOS 10.15 unknown
  274 |         std::filesystem::path path = std::filesystem::u8path(_path);
      |                                      ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:434:25: note: '~path' has been explicitly marked unavailable here
  434 |   _LIBCPP_HIDE_FROM_ABI ~path() = default;
      |                         ^
[...]/fluidsynth-2.5.0/src/utils/fluid_sys_cpp11.cpp:274:31: error: '~path' is unavailable: introduced in macOS 10.15 unknown
  274 |         std::filesystem::path path = std::filesystem::u8path(_path);
      |                               ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:434:25: note: '~path' has been explicitly marked unavailable here
  434 |   _LIBCPP_HIDE_FROM_ABI ~path() = default;
      |                         ^
[...]/fluidsynth-2.5.0/src/utils/fluid_sys_cpp11.cpp:275:39: error: 'last_write_time' is unavailable: introduced in macOS 10.15 unknown
  275 |         auto mtime = std::filesystem::last_write_time(path).time_since_epoch();
      |                                       ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/operations.h:215:45: note: 'last_write_time' has been explicitly marked unavailable here
  215 | inline _LIBCPP_HIDE_FROM_ABI file_time_type last_write_time(const path& __p) { return __last_write_time(__p); }
      |                                             ^
[...]/fluidsynth-2.5.0/src/utils/fluid_file.cpp:34:26: error: 'path' is unavailable: introduced in macOS 10.15 unknown
   34 |         std::filesystem::path _path = std::filesystem::u8path(path);
      |                          ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:382:33: note: 'path' has been explicitly marked unavailable here
  382 | class _LIBCPP_EXPORTED_FROM_ABI path {
      |                                 ^
[...]/fluidsynth-2.5.0/src/utils/fluid_file.cpp:34:56: error: 'u8path<const char *, 0>' is unavailable: introduced in macOS 10.15 unknown
   34 |         std::filesystem::path _path = std::filesystem::u8path(path);
      |                                                        ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/u8path.h:78:60: note: 'u8path<const char *, 0>' has been explicitly marked unavailable here
   78 | _LIBCPP_HIDE_FROM_ABI _LIBCPP_DEPRECATED_WITH_CHAR8_T path u8path(const _Source& __s) {
      |                                                            ^
[...]/fluidsynth-2.5.0/src/utils/fluid_file.cpp:34:39: error: '~path' is unavailable: introduced in macOS 10.15 unknown
   34 |         std::filesystem::path _path = std::filesystem::u8path(path);
      |                                       ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:434:25: note: '~path' has been explicitly marked unavailable here
  434 |   _LIBCPP_HIDE_FROM_ABI ~path() = default;
      |                         ^
[...]/fluidsynth-2.5.0/src/utils/fluid_file.cpp:34:31: error: '~path' is unavailable: introduced in macOS 10.15 unknown
   34 |         std::filesystem::path _path = std::filesystem::u8path(path);
      |                               ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/path.h:434:25: note: '~path' has been explicitly marked unavailable here
  434 |   _LIBCPP_HIDE_FROM_ABI ~path() = default;
      |                         ^
[...]/fluidsynth-2.5.0/src/utils/fluid_file.cpp:36:37: error: 'exists' is unavailable: introduced in macOS 10.15 unknown
   36 |             return std::filesystem::exists(_path);
      |                                     ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/operations.h:151:35: note: 'exists' has been explicitly marked unavailable here
  151 | inline _LIBCPP_HIDE_FROM_ABI bool exists(const path& __p) { return exists(__status(__p)); }
      |                                   ^
[...]/fluidsynth-2.5.0/src/utils/fluid_file.cpp:38:37: error: 'is_regular_file' is unavailable: introduced in macOS 10.15 unknown
   38 |             return std::filesystem::is_regular_file(_path);
      |                                     ^
/Applications/Xcode_16.4.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/c++/v1/__filesystem/operations.h:194:35: note: 'is_regular_file' has been explicitly marked unavailable here
  194 | inline _LIBCPP_HIDE_FROM_ABI bool is_regular_file(const path& __p) { return is_regular_file(__status(__p)); }
      |                                   ^
6 errors generated.
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -569,7 +569,7 @@ if ( osal STREQUAL "cpp11" )
     set( CMAKE_CXX_STANDARD 17 )
     include( CheckCXXSourceCompiles )
     check_cxx_source_compiles( "#include <filesystem>
-        int main() { std::filesystem::path path = std::filesystem::current_path(); }" HAVE_CXX_FILESYSTEM )
+        int main() { static_assert(__cpp_lib_filesystem == 201703L); std::filesystem::path p = std::filesystem::current_path(); }" HAVE_CXX_FILESYSTEM )
     if ( NOT HAVE_CXX_FILESYSTEM )
         message ( WARNING "C++ filesystem support not found, some file operations will be stubs" )
     endif ( NOT HAVE_CXX_FILESYSTEM )

